meta-resin-common/recipes-support/resin-conf-reset/resin-conf-reset/resin-conf-reset

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+set -e
+
+TARGET_GROUP=resin-bind.target
+RESIN_CONF_MP=@RESIN_CONF_MP@
+
+# Takes 2 arguments (What, Where)
+populate() {
+    # Re-construct directory structure and populate files
+    if [ -f $2 ]; then
+        mkdir -p "$(dirname $1)" && cp $2 $1
+    else
+        mkdir -p $1 && find $2 -type f -exec cp {} $1 +
+    fi
+}
+
+# Create rollback data
+mv ${RESIN_CONF_MP}/root-overlay ${RESIN_CONF_MP}/rollback || (echo "[ERROR] Unable to backup data. Refusing to reset partition" ; exit 1)
+
+# Purge data
+rm -rf /mnt/conf/root-overlay
+
+# Re-populate resin-conf with factory state
+MLIST=$(find /etc/systemd/system/${TARGET_GROUP}.wants -name "*.mount")
+
+IFS=' ' read -ra mounts <<< $MLIST
+
+for m in "${mounts[@]}"
+do
+    WHAT="$(systemctl show "$(basename "$(readlink -f $m)")" -p What | cut -d '=' -f 2)"
+    WHERE="$(systemctl show "$(basename "$(readlink -f $m)")" -p Where | cut -d '=' -f 2)"
```

Comment:
This method feels very very fragile. Have we considered other altenatives? We can also drop the reset feature and re-introduce it after the release